### PR TITLE
Testing part1

### PR DESCRIPTION
**Pull request by ishpaul777**
## In this Pull request Me and my coding partner @hammaazarok  Implemented the following-
- Use jest framework for testing.
- Created a test file [crud.test.js] to test the Add or Remove task to list in UI AND local Storage Functions 
## **Updated Testing Part 2 code added see after the Approval of test part1**
Following tests are written 
- checks for function for editing the task description.
- checks for a function for updating an item's 'completed' status.
- checks for the "Clear all completed" function.

### CODE REVIEW

File: src/tests/crud.test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { Task, Store, UI } from '../add_remove_update.js';
+
+// delete window.location;
+// window.location = {
+//     reload: jest.fn()
+// }
+// ? testing the adding function
```
Comment (OmarRamoun) [inline src/tests/crud.test.js:7]: - [ ] [REQUIRED] Please, make sure to not include any commented code becasue it pollutes the code base and makes it harder to maintain.

**Review by OmarRamoun — CHANGES_REQUESTED** [review]
# 🟡 Almost There! 🏃

## Hi @hammaazarok and @ishpaul777 👋

Good job so far! 👏. Your Coding Style is Impressive! 💪.

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃

## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** 🌟 that can improve your code quality 💯 and help you **get a better job** 💼.

_I will guide 📔 you through fixing 🔧 these bugs 🐛, just **follow my comments down below** 🔽 and make sure you **don't miss any** of them, in order to get your project approved ✅_

## Highlights 🌟

- ✔️ Descriptive commmit messsages
- ✔️ Descriptive PR description
- ✔️ No linter Errors
- :heavy_check_mark: Super Cool UI with Dark Mode. :clap: :+1: 
- The code is top-notch! 💯 and have a huge potential.
> **Note**: The UI is so cool. You have done such a great work @hammaazarok  and @ishpaul777. Keep up the great work.

## Required Changes ♻️

_Check the comments under the review._

Cheers and Happy coding!👏👏👏

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.

## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@omarramoun](https://github.com/omarramoun)
- This is My **Twitter**: [@omarramoun](https://twitter.com/omarramoun)
- This is My **Slack**: [Ramoun](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
### **Hi @ishpaul777,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi @ishpaul777 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
Linters are passing 👍 
Project is running successfully 👍 

## Required Changes ♻️
I have tried to run your test, but seemed like jest is somehow configured in wrong way, please fix so that I can run the test successfly 
image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @ishpaul777                :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements all your tests pass! good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**